irc_bot.rb

```diff
@@ -1,18 +1,120 @@
 # customize the name of your bot on the following line
-@name = "Sarah-Bot"
+require_relative './irc_bot_methods.rb'
 
+@name = "EquineVetBot"
+
+@chosen_problem = :none
+@tree_level = 0
+@repetition_lameness = 0
+@horse_general_information
+@lameness_diagnostic_program = 0
+@lameness_diagnostic_state
 # This method gets called, whenever a message is sent to our IRC channel. In it you can react to
 # the users' inputs in whatever way you like...
+
+
+
+
+
 def handle_channel_message(message)
+  if message.include?("return")
+    @tree_level = 0
+    @chosen_problem = :none
+  end
+  if @tree_level == 0
+    p message.split[3]
+    irc_send("PRIVMSG #rubymonstas :Hello and welcome to the EquineVetBot. What is your emergency?")
+    irc_send("PRIVMSG #rubymonstas :I can help you with the following frequent issues in equine medicine: Colic, Lameness, Wounds ")
+    irc_send("PRIVMSG #rubymonstas :Enter your issue")
+    irc_send("PRIVMSG #rubymonstas :If at any point of the program you want to start over, enter return.")
+    @tree_level = 1
+  end
+
+
+  #colic
+  
+  if @tree_level ==1 && message.downcase.include?("colic")
+    @chosen_problem = :colic
+  end
+  
+  if @chosen_problem == :colic
+    if @tree_level==1 && message.downcase.include?("colic")
+      handle_message_colic(message)
+      @tree_level = 2
+    end
+    if @tree_level ==2 && message.downcase.include?("yes")
```

Comment:
I would use `else if` or `case then` in this scenario for following reason:

when you set the `@tree_level = 2` above, the if-statement will match here, and the only thing that prevents entering the if is the include-string. This means, if the user inputs something like "yes, colic, no", the code of all the ifs will be run instead of just the one of the right `@tree_level`. This means, by using `else if` or `case then`, it's clearer to foresee what the code will be doing.

Reply:
so you would do one big block for colic going:
if @chosen_problem == :colic
    if @tree_level ==2 && message.downcase.include?("yes")
      colic_likely(message)
      @tree_level = 3
    elsif @tree_level ==2 && message.downcase.include?("no")
      colic_unlikely(message)
    end
    elsif @tree_level ==3 && message.downcase.include?("therapy")
      colic_therapy(message)
    end
etc?

I'm not sure I comprehend what the advantage of this is, other than a more clean-looking code? if I continue to update the @tree_level variable, it cannot run all the blocks either way, no?

ahh or maybe I just got it, would you do an if statement for each tree_level number and then between the different second conditions on the same level elsifs?

like this?
if @tree_level=1
if @tree_level=2 && xyz
elsif @tree_level=2 && snvl
elsif tree_level=2 &&slns
etc?

I know, my thinking is super complicated and I'm just SO terrible at explaining what I mean so I'm not convinced you'll know what I mean

Reply:
The advantage is that your code becomes more predictable, because you can make sure only one answer is generated for one message of the user. More predictable code makes it easier to find bugs and reconstruct occurring errors.

in the above case, if you used `if` only, it could happen that the first and third if is run, if the user includes "yes" and "therapy" in his one message. By using `elsif` you can make sure that only one if-statement is run and the others ignored.